actor/src/lib.rs

```diff
@@ -61,6 +61,16 @@ pub trait Actor {
     }
 }
 
+/// Optional trait for actors that need to perform cleanup operations during shutdown.
+/// Only actors that need cleanup should implement this trait.
+#[async_trait]
+pub trait Shutdown {
+    /// Called when the actor is shutting down to allow for cleanup operations.
+    /// This method is called after all pending messages have been processed
+    /// but before the actor loop exits.
+    async fn shutdown(&mut self);
```

Comment:
As I thought about why we used have a cache table at all (to make sure the parquet files aren't tiny), I realized this could lead to a similar result where we write smaller batches of events. For now, I think that's fine. Ideally, we're able to sort out making the aggregator idempotent on resume.

Reply:
The shutdown write will always be smaller than the 10k cache limit of course, but a full batch write yields parquet files only between 25-62 events long. I wouldn't think it makes much of a difference tbh if there are some that are even shorter if it means the aggregator being reliable on restart :grin: 

I plotted histograms of the distributions of an automatic 10k flush, and a shutdown flush of about 4k events:
<img width="1727" height="536" alt="image" src="https://github.com/user-attachments/assets/848843a7-4a9b-4953-afef-9a430dabbe2e" />